plugin/pkg/shoot/validator/admission.go

```diff
@@ -630,22 +630,26 @@ func (c *validationContext) validateProvider(a admission.Attributes) field.Error
 		if ok, minSize := validateVolumeSize(c.cloudProfile.Spec.VolumeTypes, c.cloudProfile.Spec.MachineTypes, worker.Machine.Type, worker.Volume); !ok {
 			allErrs = append(allErrs, field.Invalid(idxPath.Child("volume", "size"), worker.Volume.VolumeSize, fmt.Sprintf("size must be >= %s", minSize)))
 		}
-
-		if worker.Kubernetes != nil && worker.Kubernetes.Version != nil {
-			oldWorkerKubernetesVersion := c.oldShoot.Spec.Kubernetes.Version
-			if oldWorker.Kubernetes != nil && oldWorker.Kubernetes.Version != nil {
-				oldWorkerKubernetesVersion = *oldWorker.Kubernetes.Version
+		if worker.Kubernetes != nil {
+			if worker.Kubernetes.Kubelet != nil {
+				allErrs = append(allErrs, validateKubeletConfig(idxPath.Child("kubernetes").Child("kubelet"), c.cloudProfile.Spec.MachineTypes, worker.Machine.Type, worker.Kubernetes.Kubelet)...)
```

Comment:
There is also the following case - the kubeletConfig can be also specified on a global level under `.spec.kubernetes.kubeletConfig`. For now in this PR we only validate the kubeletConfig that is specified on Worker pool level. We also have to consider the global kubeletConfig - IIRC the pattern is that kubeletConfig can be specified on a global level and can be overwritten on worker pool level (otherwise the global kubeletConfig values will apply).

Reply:
Yes, will do, thanks!